ocaml/idl/datamodel.ml

```diff
@@ -5249,14 +5249,46 @@ let network_detach_for_vm = call
     ~allowed_roles:_R_VM_POWER_ADMIN
     ()
 
+let network_purpose = Enum ("network_purpose", [
+  "nbd", "Network Block Device service using TLS";
+  "insecure_nbd", "Network Block Device service without integrity or confidentiality: NOT RECOMMENDED";
+  "management", "Management";
+  "storage", "Storage";
+  "guest", "Guest virtual machines";
+  "himn", "Host internal management network on which guests will be assigned a private link-local IP address which can be used to talk XenAPI";
+  "unmanaged", "Not managed by xapi";
+])
+(*
+let network_add_purpose = call
+  ~name:"add_purpose"
```

Comment:
The autogenerated version (if this were a RW field) would be `add_purposes`, since the name of the field is `purposes`. I think there is value in being consistent. Given that `add_purposes` is odd, since it only adds one purpose, perhaps the field name should be `network.purpose` (this doesn't preclude a set I think)?

Reply:
> (this doesn't preclude a set I think)?

I think it precludes a set unless you mentally define "purpose" as meaning zero or more concrete purposes. I think calling it "purposes" is a lot clearer. All three options are bad:
* `Network.purpose` which is a set of purposes
* `add_purposes` which adds just one
* `add_purpose` which differs from what the autogenerated equivalent would be

I don't think I have a clear opinion on which is least objectionable.

Reply:
I'm thinking:

* "The purpose of this network is supporting connections over NBD as well as XenAPI management traffic".
* "This network has no purpose".

Reply:
Right, so
> you mentally define "purpose" as meaning zero or more concrete purposes.

I'll change the fieldname to "purpose" then (and the param-name for the constructor etc.)